### ISSUE TITLE
Use with Eleventy and relative paths in Markdown files

### ISSUE THREAD

**Issue by nhoizey** [boyum/markdown-it-image-size#404]
I'm trying to use `markdown-it-image-size` in my [Eleventy](https://11ty.dev/) projects, where 
- Markdown files are in sub folders
- Markdown code for images is relative (no path at all when the image is in the same folder as the Markdown file)

For example with the `pack11ty` project, there is a Markdown file in [this sub folder](https://github.com/nhoizey/pack11ty/tree/master/src/collections/news/2020/04/14/layout-and-design) with this content:

```markdown
---
title: Pack11ty now has a clean layout and design
date: 2020-04-14 22:38:27 +02:00
description: Pack11ty looks better with Every Layout!
tags: [layout, design]
---

Pack11ty now uses some of [Every Layout](https://every-layout.dev/) layouts and starts looking much better!

![Pack11ty screenshot](pack11ty-screenshot.png "Pack11ty's design as of 14th April 2020")
```

The image file `pack11ty-screenshot.png` is in the same folder.

However, `markdown-it-image-size` doesn't work, it doesn't find the image, which it's trying to find in the root of the project's folder:

```shell
markdown-it-image-size: Could not get dimensions of image with url pack11ty-screenshot.png Error: ENOENT: no such file or directory, open '/Users/nicolashoizey/Devs/eleventy/pack11ty/pack11ty-screenshot.png'
    at Object.openSync (node:fs:590:3)
    at syncFileToBuffer (/Users/nicolashoizey/Devs/eleventy/eleventy-plugin-pack11ty/node_modules/markdown-it-image-size/dist/index.js:1325:28)
    at imageSize (/Users/nicolashoizey/Devs/eleventy/eleventy-plugin-pack11ty/node_modules/markdown-it-image-size/dist/index.js:1364:25)
    at getImageDimensions (/Users/nicolashoizey/Devs/eleventy/eleventy-plugin-pack11ty/node_modules/markdown-it-image-size/dist/index.js:1418:18)
    at Object.md.renderer.rules.image (/Users/nicolashoizey/Devs/eleventy/eleventy-plugin-pack11ty/node_modules/markdown-it-image-size/dist/index.js:1393:15)
    at Renderer.renderInline (/Users/nicolashoizey/Devs/eleventy/eleventy-plugin-pack11ty/node_modules/markdown-it/lib/renderer.js:275:28)
    at Renderer.render (/Users/nicolashoizey/Devs/eleventy/eleventy-plugin-pack11ty/node_modules/markdown-it/lib/renderer.js:330:22)
    at MarkdownIt.render (/Users/nicolashoizey/Devs/eleventy/eleventy-plugin-pack11ty/node_modules/markdown-it/lib/index.js:544:24)
    at /Users/nicolashoizey/Devs/eleventy/pack11ty/node_modules/@11ty/eleventy/src/Engines/Markdown.js:79:38
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/Users/nicolashoizey/Devs/eleventy/pack11ty/pack11ty-screenshot.png'
}
```

Is there a way to tell `markdown-it-image-size` to find the image file in a path relative to the Markdown file path (here the same folder)?

**Comment by boyum** [comment]
MarkdownIt usually doesn't know about paths, as the render method only accepts file contents, but I'll gladly update the plugin if 11ty sends extra meta information! Let me know if your 11ty issue gets any promising answers!

**Comment by nhoizey** [comment]
It looks like there is a solution described here:
https://github.com/11ty/eleventy/issues/1510#issuecomment-1046128400

**Comment by boyum** [comment]
Nice! Fixed in @13.3.0

**Comment by nhoizey** [comment]
@boyum wow, it works perfectly, thank you so much! 😍🙏
